logfire/_internal/main.py

```diff
@@ -1898,8 +1896,7 @@ def __getattr__(self, name: str) -> Any:
 
     def __enter__(self) -> LogfireSpan:
         with handle_internal_errors():
-            self.end_on_exit = True
-            if self._span is None:
+            if self._span is None:  # pragma: no branch
```

Comment:
I think this line could be removed (and the no branch dropped), and just unconditionally set `self._span`. Wasn't sure if it was worth doing that too though at this time, given this code is likely to be refactored soon anyway.

I mostly wanted to open this PR now because of the chance that I was misunderstanding the importance of this API and it needed to be accounted for in some way while attempting to refactor